beacon-chain/rpc/service_test.go

```diff
@@ -262,9 +276,10 @@ func TestLatestAttestation(t *testing.T) {
 		<-exitRoutine
 	}(t)
 	rpcService.incomingAttestation <- attestation
-	testutil.AssertLogsContain(t, hook, "Sending attestation to RPC clients")
```

Comment:
Same here. The assertion needs to come _after_ `exitRoutine <- true`. 

Another issue is that this test originally combined two tests that should have been separated.

```go
func TestBadExample(t *testing.T) {
  go func() {
    testSomething(testChan)
    <-exitRoutine
  }
  testChan <- struct{}{}
  assertSomething()

  go func() {
    testSomething(testChan)
    <-exitRoutine
  }
  testChan <- struct{}{}
  exitRoutine <- true
  assertSomethingElse() 
}
```

The above is a race condition on `assertSomething()`. The fix is to separate into two tests:

```go
func TestSomething(t *testing.T) {
  go func() {
    testSomething(testChan)
    <-exitRoutine
  }
  testChan <- struct{}{}
  exitRountine <- true
  assertSomething()
}

fun TestSomethingElse(t *testing.T){
  go func() {
    testSomething(testChan)
    <-exitRoutine
  }
  testChan <- struct{}{}
  exitRoutine <- true
  assertSomethingElse() 
}
```

In general, each test should assert one case anyways. Long tests that sequentially assert multiple cases are hard to read/debug.